src/plugins/intel_cpu/src/nodes/mkldnn_matmul_node.cpp

```diff
@@ -22,9 +22,357 @@
 #include "mkldnn_extension_utils.h"
 #include <common/primitive_hashing_utils.hpp>
 
-using namespace mkldnn;
+#include "emitters/jit_emitter.hpp"
+#include "emitters/jit_eltwise_emitters.hpp"
+#include "emitters/jit_mkldnn_emitters.hpp"
+#include "emitters/jit_load_store_emitters.hpp"
+
 using namespace MKLDNNPlugin;
 using namespace InferenceEngine;
+using namespace mkldnn;
+using namespace mkldnn::impl::utils;
+using namespace mkldnn::impl::cpu;
+using namespace mkldnn::impl::cpu::x64;
+using namespace Xbyak;
+
+#define GET_OFF(field) offsetof(jit_matmul_args, field)
+
+
+template <cpu_isa_t isa>
+struct jit_uni_matmul_kernel_f32 : public jit_uni_matmul_kernel, public jit_generator {
+    DECLARE_CPU_JIT_AUX_FUNCTIONS(jit_uni_matmul_kernel_f32)
+
+    explicit jit_uni_matmul_kernel_f32(jit_matmul_config_params jcp_, const mkldnn_primitive_attr &attr) : jit_uni_matmul_kernel(jcp_, attr), jit_generator() {}
+
+    void create_ker() override {
+        jit_generator::create_kernel();
+        ker_ = (decltype(ker_))jit_ker();
+    }
+
+    void generate() override {
+        const auto &p = attr_.post_ops_;
+        for (int i = 0; i < p.len(); i++) {
+            auto &post_op = p.entry_[i];
+            if (post_op.is_eltwise()) {
+                eltwise_injectors.push_back(std::make_shared<jit_uni_eltwise_injector_f32<isa>>(
+                        this,
+                        post_op.eltwise.alg,
+                        post_op.eltwise.alpha,
+                        post_op.eltwise.beta,
+                        1));
+            } else {
+                IE_THROW() << "MatMul supports only eltwise post ops!";
+            }
+        }
+
+        load_emitter.reset(new jit_load_emitter(this, isa));
+        store_emitter.reset(new jit_store_emitter(this, isa));
+
+        this->preamble();
+
+        mov(reg_src_0, ptr[reg_params + GET_OFF(src0)]);
+        mov(reg_src_1, ptr[reg_params + GET_OFF(src1)]);
+        mov(reg_dst, ptr[reg_params + GET_OFF(dst)]);
+
+        load_pool_gpr_idxs = {static_cast<size_t>(reg_load_store_mask.getIdx()), static_cast<size_t>(reg_load_table.getIdx())};
+        store_pool_gpr_idxs = {static_cast<size_t>(reg_load_store_mask.getIdx())};
+        store_pool_vec_idxs = {static_cast<size_t>(vmm_zero.getIdx())};
+
+        amount_full = (jcp_.scalar_product ? jcp_.k : jcp_.n) / vec_step;
+        amount_tail = (jcp_.scalar_product ? jcp_.k : jcp_.n) % vec_step;
+
+        Xbyak::Label label_batch;
+        Xbyak::Label label_batch_end;
+
+        mov(batch, 0);
+        L(label_batch); {
+            cmp(batch, jcp_.b);
+            je(label_batch_end, T_NEAR);
+
+            body();
+
+            // strides for batch
+            if (jcp_.stride0 == 0)  // ptr for the first matrix is automatically incremented after each iteration by m,
```

Comment:
`!=0` ?
Seems like this case is not covered by tests?

Reply:
Discussed offline. d2de7f83ca4b824b91c397165a75156308b87992

It works before because for these cases no custom matmul has been selected. I fixed [condition](https://github.com/a-sidorova/openvino/blob/16d48ae9ea2955d09b618dd12f311d8c9ce00823/src/plugins/intel_cpu/src/nodes/mkldnn_matmul_node.cpp#L867) for it